Dockerfile

```diff
@@ -21,22 +21,29 @@ RUN apk update && \
         proj-dev \
         make
 
-COPY requirements.txt .
-
-RUN pip install \
-    --prefix=/install \
-    --no-cache-dir \
-    --no-warn-script-location \
-    -r requirements.txt
+# Setup backend Python dependencies
+COPY pyproject.toml pdm.lock README.md ./
+RUN pip install --no-cache-dir --upgrade pip \
+    && pip install --no-cache-dir pdm==2.0.2 \
+    && pdm config python.use_venv false
+RUN pdm install --prod --no-editable
 
 # Setup backend runtime dependencies
 FROM base
 
+WORKDIR /usr/src/app
+
+ENV PATH="/usr/src/python/bin:$PATH" \
+    PYTHONPATH="/usr/src/python/lib"
+
+# Setup backend runtime dependencies
 RUN apk update && \
-    apk add \
+    apk add --no-cache \
         postgresql-libs geos proj-util
 
-COPY --from=builder /install /usr/local
+COPY --from=builder \
+    /install/__pypackages__/3.8 \
```

Comment:
In concept, this is okay. But when we upgrade out of Python 3.7, we might as well go to Python 3.10. In for a penny, in for a pound.

Reply:
I mention this because there could be further conflicts when this is merged, eventually after the current conflicts are resolved. This will depend on my work on the CI pipelines.

Reply:
Agreed 👍 The version should probably set as a build ARG

Reply:
That will be better.

Reply:
If I may, I will update the Dockerfile a bit and base it on 3.10-alpine3.17.

Reply:
New issue for this: https://github.com/hotosm/tasking-manager/issues/5788

Can we continue the discussion there?